sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -150,8 +161,24 @@ class RocksDB(
           metadata.numKeys
         }
         numKeysOnWritingVersion = numKeys
-        numKeysOnLoadedVersion = numKeys
 
+        for (v <- latestSnapshotVersion + 1 to version) {
```

Comment:
Do we need this logic if changelog checkpointing is disabled too ?

Reply:
It will be a noop without changelog checkpointing because latest snapshot will be loaded successfully.

Reply:
Probably still better to just embed in an `if` condition for better readability ?